### PR TITLE
Row UDF scalar arguments

### PR DESCRIPTION
**Pull request by brandon-b-miller**
Closes https://github.com/dask-contrib/dask-sql/issues/279

### CODE REVIEW

**Comment by brandon-b-miller** [comment]
This is ready for review - I thought some code might need to be written to disambiguate when one has a column with a name that could also be a constant, like having a column with name `42` and also passing 42 as a constant parameter. But it turns out we can just write `\"42\"` in the query to get the column with that name instead.

File: dask_sql/datacontainer.py [hunk 1]
```diff
@@ -205,10 +205,19 @@ def __init__(self, func, row_udf: bool, return_type=None):
...
+            column_args = []
+            scalar_args = []
+            for operand in args:
+                if isinstance(operand, dd.Series):
+                    column_args.append(operand)
+                else:
+                    scalar_args.append(operand)
+                df = column_args[0].to_frame()
+                for col in column_args[1:]:
+                    df[col.name] = operand
```
Comment (ayushdg) [inline dask_sql/datacontainer.py:217]:
```suggestion
                    df[col.name] = col
```

Comment (ayushdg) [inline dask_sql/datacontainer.py:217]: I'm guessing this section was meant to be outside the loop.

**Review by ayushdg — CHANGES_REQUESTED** [review]
Thanks a lot for the changes @brandon-b-miller !
Couple of minor comments in addition to the changes suggested:
Would it be possible to also update the docstring to include an example with scalar values?

Also I was able to catch the issues below while testing this with udfs that accept multiple column args, since in the single arg case the output would be correct. Might be worth adding that case to the tests as well.

**Comment by brandon-b-miller** [comment]
Updated! :)

File: tests/integration/test_function.py [hunk 2]
```diff
@@ -63,6 +65,64 @@ def f(row):
...
+
+    statement = f"SELECT F(a, {k}) as a from df"
+
+    return_df = c.sql(statement)
+    return_df = return_df.compute()
+    expectation = op(df[["a"]], k).astype(retty)
+    assert_frame_equal(return_df.reset_index(drop=True), expectation)
+
+
+# Test row UDFs with one args
```
Comment (charlesbluca) [inline tests/integration/test_function.py:92]:
```suggestion
# Test row UDFs with two args
```

File: tests/integration/test_function.py [hunk 3]
```diff
@@ -63,6 +65,64 @@ def f(row):
     assert_frame_equal(return_df.reset_index(drop=True), expectation)
 
 
+# Test row UDFs with one args
```
Comment (charlesbluca) [inline tests/integration/test_function.py:68]:
```suggestion
# Test row UDFs with one arg
```

**Review by charlesbluca — COMMENTED** [review]
Thanks for the work here @brandon-b-miller 🙂 just small nitpicks, the code itself LGTM

File: dask_sql/datacontainer.py [hunk 4]
```diff
@@ -205,10 +205,19 @@ def __init__(self, func, row_udf: bool, return_type=None):
...
-                df[operand.name] = operand
-            result = df.apply(self.func, axis=1, meta=self.meta).astype(self.meta[1])
+            column_args = []
+            scalar_args = []
+            for operand in args:
+                if isinstance(operand, dd.Series):
+                    column_args.append(operand)
+                else:
+                    scalar_args.append(operand)
+                df = column_args[0].to_frame()
```
Comment (ayushdg) [inline dask_sql/datacontainer.py:215]:
```suggestion
            df = column_args[0].to_frame()
```

**Review by ayushdg — APPROVED** [review]
One small suggestion otherwise lgtm!

**Review by charlesbluca — APPROVED** [review]
LGTM; thanks for the work @brandon-b-miller 😄

**Pull request merged**